### ISSUE TITLE
custom op Devel_Peek_Dump with -O1

### ISSUE THREAD

**Issue by atoomic** [rurban/perl-compiler#274]
This came with 5.19.3

steps to reproduce:
```sh
> perl -e 'use Devel::Peek; my %hash = ( a => 1 ); Dump(%hash) if $ENV{FALSE}; print "ok\n";'
ok
> rm -f a.out*; perlcc -r -e 'use Devel::Peek; my %hash = ( a => 1 ); Dump(%hash) if $ENV{FALSE}; print "ok\n";'
ok
> rm -f a.out*; perlcc -O1 -r -e 'use Devel::Peek; my %hash = ( a => 1 ); Dump(%hash) if $ENV{FALSE}; print "ok\n";'
pccLL6Mn.c:4659: error: ‘OP_DEVEL_PEEK_DUMP’ undeclared here (not in a function)
Can't exec "./a.out": No such file or directory at /usr/local/cpanel/3rdparty/perl/522-debug/bin/perlcc line 136.
````

**Comment by atoomic** [comment]
dummy fix
```diff
diff --git a/lib/B/C.pm b/lib/B/C.pm
index 0b8c83b..e81ffcf 100644
--- a/lib/B/C.pm
+++ b/lib/B/C.pm
@@ -1135,9 +1135,14 @@ sub force_heavy {
 # in perl_init ( ~10 bytes/op with GCC/i386 )
 sub B::OP::fake_ppaddr {
   return "NULL" unless $_[0]->can('name') ;
+
+  my $ucname = uc( $_[0]->name );
+  # no optimizations for Devel::Peek
+  return "NULL" if $ucname =~ qr{DEVEL_PEEK};
+
   return $B::C::optimize_ppaddr
-    ? sprintf( "INT2PTR(void*,OP_%s)", uc( $_[0]->name ) )
-    : ( $verbose ? sprintf( "/*OP_%s*/NULL", uc( $_[0]->name ) ) : "NULL" );
+    ? sprintf( "INT2PTR(void*,OP_%s)", $ucname )
+    : ( $verbose ? sprintf( "/*OP_%s*/NULL", $ucname ) : "NULL" );
 }
 sub B::FAKEOP::fake_ppaddr { "NULL" }
 # XXX HACK! duct-taping around compiler problems
```

**Comment by rurban** [comment]
Devel::Peek 1.14 with 5.19.3 added a custom op `Devel_Peek_Dump`

**Comment by toddr** [comment]
./t/testc.sh 274

**Comment by toddr** [comment]
When I run it on travis (presumably without debug, I get this from testc 274. GCC fails to compile.

# xtestc/0274.c:9183:29: error: ‘OP_DEVEL_PEEK_DUMP’ undeclared here (not in a function)

**Comment by rurban** [comment]
Fixed with commit ab9b0eb638b8d1a0d16f40e2c3b2fd40feb03a91
Date:   Mon Nov 2 13:01:34 2015 +0100

    C 1.52_11: Support custom op Devel_Peek_Dump
